microsoft/testsuites/ltp/ltp.py

```diff
@@ -0,0 +1,272 @@
+                f" {drive_name} "
+            )
+        else:
+            parameters = f"-p -q -l {self.LTP_RESULT_PATH} -o {self.LTP_OUTPUT_PATH} "
+
+        # add the list of tests to run
+        parameters += f"-f {','.join(ltp_tests)} "
+
+        # add the list of skip tests to run
+        if len(skip_tests) > 0:
+            # write skip test to skipfile with newline separator
+            skip_file_value = "\n".join(skip_tests)
+            self.node.tools[Echo].write_to_file(
+                skip_file_value, PurePosixPath(self.LTP_SKIP_FILE), sudo=True
+            )
+            parameters += f"-S {self.LTP_SKIP_FILE} "
+
+        # run ltp tests
+        self.run(parameters, sudo=True, force_run=True, timeout=12000)
+
+        # parse results
+        result_messages = self._parse_results(
+            lisa_test_name, environment, self.LTP_RESULT_PATH
+        )
+
+        return result_messages
+
+    def _install(self) -> bool:
+        assert isinstance(self.node.os, Posix), f"{self.node.os} is not supported"
+
+        # install common dependencies
+        self.node.os.install_packages(
+            [
+                "m4",
+                "bison",
+                "flex",
+                "psmisc",
+                "autoconf",
+                "automake",
+            ]
+        )
+
+        # install distro specific dependencies
+        if isinstance(self.node.os, Fedora):
+            self.node.os.install_packages(
+                ["libaio-devel", "libattr", "libcap-devel", "libdb"]
+            )
+            if not (
+                isinstance(self.node.os, Redhat)
+                and self.node.os.information.release >= "8.0"
+            ):
+                self.node.os.install_packages(["db4-utils"])
+            else:
+                self.node.os.install_packages(["ntp"])
+        elif isinstance(self.node.os, Debian):
+            self.node.os.install_packages(
+                [
+                    "ntp",
+                    "libaio-dev",
+                    "libattr1",
+                    "libcap-dev",
+                    "keyutils",
+                    "libdb4.8",
+                    "libberkeleydb-perl",
+                    "expect",
+                    "dh-autoreconf",
+                    "gdb",
+                    "libnuma-dev",
+                    "quota",
+                    "genisoimage",
+                    "db-util",
+                    "unzip",
+                    "exfat-utils",
+                ]
+            )
+        elif isinstance(self.node.os, Suse):
+            self.node.os.install_packages(
+                [
+                    "ntp",
+                    "git-core",
+                    "db48-utils",
+                    "libaio-devel",
+                    "libattr1",
+                    "libcap-progs",
+                    "libdb-4_8",
+                    "perl-BerkeleyDB",
+                ]
+            )
```

Comment:
how about other distros?

Reply:
I have tested on the major classes: Fedore, Suse, Debian and Mariner. This was also the class of OS supported by v2.